src/Symfony/Component/JsonEncoder/Attribute/MaxDepth.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\Attribute;
+
+/**
+ * Defines the maximum encoding depth for the property.
+ *
+ * When the maximum depth is reached, a {@see \Symfony\Component\JsonEncoder\Encode\Normalizer\NormalizerInterface}
+ * service is called if $maxDepthReachedNormalizerServiceId has been defined.
+ *
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY)]
+final class MaxDepth
```

Comment:
does this need to be an attribute? shouldn't this be handled as an encoder option?